mysterious-mice/spacebook/posts/views.py

```diff
@@ -0,0 +1,94 @@
+from django.views.generic import ListView, DetailView
+from django.core.exceptions import ObjectDoesNotExist
+from .models import ImagePost
+from rovers.models import Rover
+
+
+class ImagePostListView(ListView):
+    """
+    Provides a list of ImagePosts to be displayed in a grid. If a valid username is provided as a
+    GET parameter, then the ImagePosts returned will be limited to that user. This parameter will
+    be passed automatically if this page is navigated to from a rover's profile.
+    """
+
+    model = ImagePost
+    template_name = "posts/home.html"
+    context_object_name = "posts"
+    ordering = ["-date"]
+    paginate_by = 16
+
+    # if a valid username provided, only return that rover's posts
+    def get_queryset(self):
```

Comment:
nice exception handeling. looks very pythonic. do imagepost need to be inside the try as well?